ipykernel/kernelapp.py

```diff
@@ -772,30 +772,37 @@ def initialize(self, argv=None):
         sys.stdout.flush()
         sys.stderr.flush()
 
-    def start(self) -> None:
-        """Start the application."""
+    async def _start(self) -> None:
+        """
+        Async version of start, when the loop is not controlled by IPykernel
+
+        For example to be used in test suite with @pytest.mark.trio
+        """
         if self.subapp is not None:
             self.subapp.start()
             return
         if self.poller is not None:
             self.poller.start()
```

Comment:
Does it make sense to have `_start` and `main`, or should we just have `main` and include these lines in it?

Reply:
Ah yes :+1: